proposals/solid-oidc/primer.bs

```diff
@@ -0,0 +1,853 @@
+necessary to gain access to any Pod. In this demo, Alice's OP is at
+`https://secureauth.example`.
+
+**Alice's WebID** - Alice's WebID is `https://alice.coolpod.example/profile/card#me`. It serves
+as her unique identifier.
+
+**Decent Photos (RP)** - decentphotos is a third party photo viewing web application hosted at
+`https://decentphotos.example`. This web app allows you to view your photos as well as your
+friend's photos. It will also perform cron jobs on the photos to detect faces. In the OIDC
+world, this is known as the Relying Party (RP).
+
+**RP WebID** - The RP (`https://decentphotos.example`) has a WebID of its own to list
+information about the app. Decentphoto's is at `https://decentphotos.example/webid#this`
+
+## Application Flow
+
+### Logging in with authorization code flow with PKCE
+
+<img src="primer-login.png" />
+
+#### 1. Alice uses the decentphotos app
+
+Alice has heard of a great new site that allows her to view her friend's photos and tag faces.
+She navigates to `https://decentphotos.example` via her web browser which returns an HTML page.
+This page contains JavaScript that will help with the authentication process.
+
+#### 2. Alice selects her OP or WebID
+
+Before decentphotos can start displaying images, Alice needs to start the process of providing
+consent. To do so, she must either provide her WebID
+(`https://alice.coolpod.example/profile/card#me`) or the URL of her OP
+(`https://secureauth.example`) 
+
+While it is not the case with Alice, a user's Pod and OP can be hosted at the same domain. For
+example, Bob's Pod could be `bob.otherpod.example` with a WebID of
+`https://bob.otherpod.example/profile/card#me`, but his OP is at `https://solid.example`.
+
+#### 2.1 Retrieve Profile (Only needed if a WebID is provided)
+
+If Alice entered her WebID's URL rather than her OP's URL, a request should be made to
+determine her OP. To do so, a request should be made to Alice's WebId:
+
+```bash
+GET https://alice.coolpod.example/profile/card#me
+```
+
+It will return a body similar to this:
+
+```
+@prefix : <#>.
+@prefix solid: <http://www.w3.org/ns/solid/terms#>.
+@prefix foaf: <http://xmlns.com/foaf/0.1/>.
+@prefix schema: <http://schema.org/>.
+
+<>
+    a foaf:PersonalProfileDocument ;
+    foaf:maker <https://localhost:8443/profile/card#me> ;
+    foaf:primaryTopic <https://localhost:8443/profile/card#me> .
+
+:me a foaf:Person ;
+    a schema:Person ;
+    foaf:name "Alice" ;
+    solid:oidcIssuer <https://secureauth.example> ;
+```
+
+The OP URL is located at `:me -> solid:oidcIssuer`
+
+#### 3. Retrieves OP Configuration
+
+Now that we have Alice's OP's URL, the RP must make a request to retrieve the OP's
+configuration. This is always located at the OP's hostname followed by `/.well-known
+openid-configuration`
+
+```
+GET https://secureauth.example/.well-known/openid-configuration
+```
+
+The [openid-configuration](https://auth0.com/docs/protocols/oidc/openid-connect-discovery)
```

Comment:
Since this is a spec-adjacent doc, would it make sense to link to the corresponding spec? (instead of a vendor's docs). We could link to https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata, which is where those docs come from.